contracts/root/TokenPredicates/BurnableERC1155Predicate.sol

```diff
@@ -0,0 +1,306 @@
+        uint256[] ids,
+        uint256[] amounts
+    );
+
+    constructor() public {}
+
+    function initialize(address _owner) external initializer {
+        _setupContractId("BurnableERC1155Predicate");
+        _setupRole(DEFAULT_ADMIN_ROLE, _owner);
+        _setupRole(MANAGER_ROLE, _owner);
+    }
+
+    /**
+     * @notice rejects single transfer
+     */
+    function onERC1155Received(
+        address,
+        address,
+        uint256,
+        uint256,
+        bytes calldata
+    ) external override returns (bytes4) {
+        return 0;
+    }
+
+    /**
+     * @notice accepts batch transfer
+     */
+    function onERC1155BatchReceived(
+        address,
+        address,
+        uint256[] calldata,
+        uint256[] calldata,
+        bytes calldata
+    ) external override returns (bytes4) {
+        return ERC1155Receiver(0).onERC1155BatchReceived.selector;
+    }
+
+    /**
+     * @notice Lock ERC1155 tokens for deposit, callable only by manager
+     * @param depositor Address who wants to deposit tokens
+     * @param depositReceiver Address (address) who wants to receive tokens on child chain
+     * @param rootToken Token which gets deposited
+     * @param depositData ABI encoded id array and amount array
+     */
+    function lockTokens(
+        address depositor,
+        address depositReceiver,
+        address rootToken,
+        bytes calldata depositData
+    ) external override only(MANAGER_ROLE) {
+        // forcing batch deposit since supporting both single and batch deposit introduces too much complexity
+        (
+            uint256[] memory ids,
+            uint256[] memory amounts,
+            bytes memory data
+        ) = abi.decode(depositData, (uint256[], uint256[], bytes));
+
+        emit LockedBatchBurnableERC1155(
+            depositor,
+            depositReceiver,
+            rootToken,
+            ids,
+            amounts
+        );
+        IBurnableERC1155(rootToken).safeBatchTransferFrom(
+            depositor,
+            address(this),
+            ids,
+            amounts,
+            data
+        );
+    }
+
+    /**
+     * @notice Validates log signature, from and to address
+     * then sends the correct tokenId, amount to withdrawer/ burns token, as instructed
+     * callable only by manager
+     * @param rootToken Token which gets withdrawn
+     * @param log Valid ERC1155 TransferSingle burn or TransferBatch burn log from child chain
+     */
+    function exitTokens(
+        address,
+        address rootToken,
+        bytes memory log
+    ) public override only(MANAGER_ROLE) {
+
+        RLPReader.RLPItem[] memory logRLPList = log.toRlpItem().toList();
+        RLPReader.RLPItem[] memory logTopicRLPList = logRLPList[1].toList();
+
+        bytes memory logData = logRLPList[2].toBytes();
+        address withdrawer = address(logTopicRLPList[2].toUint());
+
+        if (bytes32(logTopicRLPList[0].toUint()) == TRANSFER_SINGLE_EVENT_SIG) {
```

Comment:
Extract `bytes32(logTopicRLPList[0].toUint()` into variable and check conditions in if-else-if chain, let's not burn gas.